File: prolog/p-03.pl

element_at(N, X, [X|_]) :- N == 1, !.
element_at(_, _, []) :-
    fail, !.
element_at(N, _, [_|_]) :- N < 1, fail, !.
element_at(N, X, [_|Xs]) :-
    N1 is N - 1, element_at(N1, X, Xs), !.
			 

File: erlang/p_03.erl
-module(p_03).
-export([element_at/2]).

element_at(_, K) when K=<0 ->
    erlang:error(index_error);
element_at([], K) when K>1 ->
    erlang:error(index_out_range);
element_at([X|_], 1) ->
    X;
element_at([_|T], K) when K>1 ->
    element_at(T, K - 1).



